### DISCUSSION TITLE
Should we have entropy at Raintree level?

### DISCUSSION THREAD

**Discussion by deblasis** [pokt-network/pocket#278] [Architecture]
Tending to the TODOes:

```
./p2p/types/raintree.pb.go:	Nonce uint64 `protobuf:"varint,3,opt,name=nonce,proto3" json:"nonce,omitempty"` // DISCUSS(drewsky): discuss if we should have entropy at the RainTree level.
./p2p/raintree/types/proto/raintree.proto:  // DISCUSS(drewsky): discuss if we should have entropy at the RainTree level.
```

**Comment by Olshansk** [comment]
Some context for this discussion:
- `Nonce`s will be added into the messages constructed by modules such as consensus or utility
- **Additional** entropy can be added exclusively at the P2P/RainTree level

**Pros:**
* Extra deduping at the P2P level if a malicious/faulty node rebroadcasts (e.g. DOSses) the network at the network layer

**Cons:**
* Malicious node can reconstruct a P2P message (with the same underlying message) and a different entropy so it is processed by the network

**Goal**
* This is an extra, but not complete, layer of deduiping.

The decision to this discussion can be started here but moved to the M4 milestone,

**Comment by bryanchriswhite** [comment]
backlinking related issue:
- https://github.com/pokt-network/pocket/issues/280
